### ISSUE TITLE
S3_PATH is not cleaned correctly when listing remote backups

### ISSUE THREAD

**Issue by prskr** [Altinity/clickhouse-backup#406]
Due to how `clickhouse-backup` interacts with the S3 API when listing existing backups, the `S3_PATH` is joined twice to the backup name before fetching the `metadata.json` file resulting in something like:

```HTTP
HEAD /<bucket>/<prefix>/<prefix>/<backup-name>/metadata.json
```

This isn't an issue as long as the prefix is empty but if you try to create backups in sub-directories it's not possible to list/restore them later on.

I'd be glad to provide a fix if desired :blush:

**Comment by Slach** [comment]
Which version clickhouse-backup do you use?

could you share
`S3_DEBUG=1 clickhouse-backup list remote`
result?

**Comment by prskr** [comment]
Holy shit, that was a lot faster than I expected :smile: 

With debug enabled calling `list remote` I noticed that the  path of the HEAD request is incorrect:

```
-----------------------------------------------------
2022/03/01 16:07:49.004887  info DEBUG: Request s3/HeadObject Details:
---[ REQUEST POST-SIGN ]-----------------------------
HEAD /clickhouse-backup/my-prefix/my-prefix/2022-03-01T09-24-39/metadata.json HTTP/1.1
Host: xxxxxxxxxxx
User-Agent: aws-sdk-go/1.40.31 (go1.17.7; linux; amd64)
Authorization: xxxxxxxxxxx
X-Amz-Content-Sha256: xxxxxxx
X-Amz-Date: xxxxxxxxxxxx
```

I already created a PR #407 that resolves the issue in my tests so far.

**Comment by Slach** [comment]
could you provide results for 
```clickhouse-backup print-config``` ?

**Comment by prskr** [comment]
I'll provide the output tomorrow.
Out of curiosity: do you test against an AWS S3 endpoint or MinIO or something else?

**Comment by Slach** [comment]
We test both real S3 and minio backends

ok. will wait `print-config` to attempt to figure-out root reason
you can skip access credentials
but please don't skip any path or bucket or region settings

**Comment by prskr** [comment]
Okay, the config you were asking for is now:

```yaml
general:
  remote_storage: s3
  max_file_size: 1073741824
  disable_progress_bar: true
  backups_to_keep_local: 0
  backups_to_keep_remote: 0
  log_level: debug
  allow_empty_backups: true
  download_concurrency: 4
  upload_concurrency: 4
  restore_schema_on_cluster: ""
  upload_by_part: true
  download_by_part: true
clickhouse:
  username: default
  password: ""
  host: localhost
  port: 9000
  disk_mapping: {}
  skip_tables:
  - system.*
  - INFORMATION_SCHEMA.*
  - information_schema.*
  timeout: 5m
  freeze_by_part: false
  secure: false
  skip_verify: false
  sync_replicated_tables: false
  log_sql_queries: false
  config_dir: /etc/clickhouse-server/
  restart_command: systemctl restart clickhouse-server
  ignore_not_exists_error_during_freeze: true
  debug: false
s3:
  access_key: xxxxxxx
  secret_key: xxxxxx
  bucket: clickhouse-backup
  endpoint: https://my-minio-endpoint
  region: minio
  acl: private
  assume_role_arn: ""
  force_path_style: true
  path: my-prefix
  disable_ssl: false
  compression_level: 1
  compression_format: brotli
  sse: ""
  disable_cert_verification: true
  storage_class: STANDARD
  concurrency: 4
  part_size: 0
  debug: true
gcs:
  credentials_file: ""
  credentials_json: ""
  bucket: ""
  path: ""
  compression_level: 1
  compression_format: tar
  debug: false
  endpoint: ""
cos:
  url: ""
  timeout: 2m
  secret_id: ""
  secret_key: ""
  path: ""
  compression_format: tar
  compression_level: 1
  debug: false
api:
  listen: 0.0.0.0:7171
  enable_metrics: true
  enable_pprof: false
  username: ""
  password: ""
  secure: false
  certificate_file: ""
  private_key_file: ""
  create_integration_tables: true
  allow_parallel: false
ftp:
  address: ""
  timeout: 2m
  username: ""
  password: ""
  tls: false
  path: ""
  compression_format: tar
  compression_level: 1
  concurrency: 8
  debug: false
sftp:
  address: ""
  port: 22
  username: ""
  password: ""
  key: ""
  path: ""
  compression_format: tar
  compression_level: 1
  concurrency: 1
  debug: false
azblob:
  endpoint_suffix: core.windows.net
  account_name: ""
  account_key: ""
  sas: ""
  use_managed_identity: false
  container: ""
  path: ""
  compression_level: 1
  compression_format: tar
  sse_key: ""
  buffer_size: 0
  buffer_count: 3
```

I guess the original problem I had was that I set `S3_PATH` to `/my-prefix` instead of `my-prefix` which led to the wrong path prefix trimming.
Which also explains my PR breaks your behavior because I assumed there should be a leading forward slash (and I also tested it like that).

Of course I could add this case in the PR but I think a better approach would be to validate this when the S3 storage backend is initialized and fix the prefix if required directly there?
Any thoughts?

**Comment by Slach** [comment]
ok. thanks for clarify, i'll fix issue on next 1.3.2 release
